### PR TITLE
timeout for databricks

### PR DESCRIPTION
**Pull request by y-f-u**
this will address one of the problem in #1033 .

Default timeout is 30s, which failed on my test - a query with where clause on 30G+ dataset.
```
sql> select * from transactions where hash = '0xeabc0e3694a6e202c360a48b380b4205ee15541b816d66f56da687b3a3a0fd79';
Error An internal error occurred while processing the query. Show technical details with '.error'
No data returned for query
```

Increase it to 120s fixed the issue, result is returned successfully.

By
```
params:
  timeout: 120s
```

```
sql> select * from transactions where hash = '0xeabc0e3694a6e202c360a48b380b4205ee15541b816d66f56da687b3a3a0fd79';



+--------------------------------------------------------------------+-------+-------------------+--------------------------------------------+--------------------------------------------+-------+-------------+-------+-----------------------------+------------------+--------------------------+--------------+----------------+-----------------------+--------------+--------------------------------------------------------------------+-----------------+--------------------------+------------------+-----------------------------+-----------+
| hash                                                               | nonce | transaction_index | from_address                               | to_address                                 | gas   | gas_price   | input | receipt_cumulative_gas_used | receipt_gas_used | receipt_contract_address | receipt_root | receipt_status | block_timestamp       | block_number | block_hash                                                         | max_fee_per_gas | max_priority_fee_per_gas | transaction_type | receipt_effective_gas_price | end_block |
+--------------------------------------------------------------------+-------+-------------------+--------------------------------------------+--------------------------------------------+-------+-------------+-------+-----------------------------+------------------+--------------------------+--------------+----------------+-----------------------+--------------+--------------------------------------------------------------------+-----------------+--------------------------+------------------+-----------------------------+-----------+
| 0xeabc0e3694a6e202c360a48b380b4205ee15541b816d66f56da687b3a3a0fd79 | 18    | 74                | 0xdc4392cfde1d0e467a3d7602bbbc500bfe20304d | 0xdc4392cfde1d0e467a3d7602bbbc500bfe20304d | 46814 | 90340839758 | input | 3950477                     | 21000            |                          |              | 1              | +54048-12-19T15:20:00 | 14100236     | 0x000d2ed73601ebe1cadd19bec68cf2e59a44239bf8ee29e43da801d0d49fa2ed | 110709536502    | 1650000000               | 2                | 90340839758                 | 14441014  |
+--------------------------------------------------------------------+-------+-------------------+--------------------------------------------+--------------------------------------------+-------+-------------+-------+-----------------------------+------------------+--------------------------+--------------+----------------+-----------------------+--------------+--------------------------------------------------------------------+-----------------+--------------------------+------------------+-----------------------------+-----------+

Query took: 1008.800289729 seconds. 1/1 rows displayed.
```

### CODE REVIEW

File: crates/runtime/src/dataconnector/databricks.rs [hunk 1]
```diff
@@ -172,6 +177,15 @@ async fn get_delta_table(
     };
     storage_options.insert(AWS_S3_ALLOW_UNSAFE_RENAME.to_string(), "true".to_string());
 
+    let timeout: Option<String> = params
+        .as_ref() // &Option<HashMap<String, String>>
+        .as_ref() // Option<&HashMap<String, String>>
+        .and_then(|params| params.get("timeout").cloned());
+
+    if let Some(timeout) = timeout {
+        storage_options.insert("timeout".to_string(), timeout);
```
Comment (phillipleblanc) [inline crates/runtime/src/dataconnector/databricks.rs:188]: What happens if the timeout value is invalid?

Comment (phillipleblanc) [reply crates/runtime/src/dataconnector/databricks.rs:188]: Let's increase the default to 60s

Comment (phillipleblanc) [reply crates/runtime/src/dataconnector/databricks.rs:188]: Do we need this option for S3 as well?

Comment (y-f-u) [reply crates/runtime/src/dataconnector/databricks.rs:188]: > Let's increase the default to 60s

60s is not the magic number. It's rather good to stick the downstream dependency is, and leave the user to override it. Which follows the same pattern delta-rs using object storage.

> Do we need this option for S3 as well?

No need for s3 so far from the testing.

Comment (y-f-u) [reply crates/runtime/src/dataconnector/databricks.rs:188]: > failed to parse "60xxljksd" as Duration

Error propagated from delta-rs.

**Comment by ablyler** [comment]
I went ahead and checked out this branch and ran it with the following in the dataset.yaml:

```yaml
params:
  timeout: 300s
```

I got the following when trying to run a query with a where clause (that matches a liquid clustered column):

```
sql> select * from TABLE_NAME where ip = '1.1.1.1';
Error An internal error occurred while processing the query. Show technical details with '.error'
No data returned for query
```

```
sql> .error
Status { code: Internal, message: "Parquet error: Parquet error: AsyncChunkReader::get_bytes error: Generic S3 error: Error after 1 retries in 240.140585625s, max_retries:10, retry_timeout:180s, source:error sending request for url (https://s3.us-east-1.amazonaws.com/BUCKET/CATALOG/__unitystorage/schemas/GUID/tables/GUID/part-GUID.c000.snappy.parquet): operation timed out", source: None }
```

**Comment by y-f-u** [comment]
@ablyler thanks for trying this out. I think the timeout does resolve one issue, but we are likely bumping into another issue described here https://github.com/delta-io/delta-rs/issues/2301.

I will get this partial fix merged. Meanwhile, will investigate the new symptom.

File: crates/runtime/src/dataconnector/databricks.rs (hunk 1 above)
Comment (y-f-u) [reply crates/runtime/src/dataconnector/databricks.rs:188]: set the default to 300s

**Review by digadeesh — APPROVED** [review]

**Pull request merged**